platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/util/MarkerFileWriter.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.util;
+
+import com.swirlds.common.context.PlatformContext;
+import com.swirlds.logging.legacy.LogMarker;
+import com.swirlds.platform.config.PathsConfig;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Writes marker files with the given filename to disk in the configured directory.  If there is no configured
+ * directory, no marker files are written.
+ */
+public class MarkerFileWriter {
+
+    /**
+     * The logger for this class.
+     */
+    private static final Logger LOG = LogManager.getLogger(MarkerFileWriter.class);
+
+    /**
+     * The directory where the marker files are written.  If null, no marker files are written.
+     */
+    private final File markerFileDirectory;
+
+    /**
+     * Creates a new {@link MarkerFileWriter} with the given {@link PlatformContext}.  If the marker file writer is
+     * enabled, files are written to the configured directory.
+     *
+     * @param platformContext the platform context containing configuration.
+     */
+    public MarkerFileWriter(@NonNull final PlatformContext platformContext) {
```

Comment:
There's multiple aesthetics at play.  I find the uniformity of passing in a `PlatformContext` into constructors to be very clean and the asymmetry of sometimes passing in the `PlatformContext` and sometimes passing in specific pieces or configuration to be less clean.  

Since the `MarkerFileWriter` is constructed in multiple locations, `IssDetector` and `ConsensusImpl` for now, but there could be more locations in the future, there would be a multiplicity of locations where the file location is extracted from the `PlatformContext`.  Should we in the future change where the config value is located, then each location that extracts the directory from the `PlatformContext` would need to be modified.

Reply:
just added a rate limited logger which uses time from the platform context.